transport/rif_comms/node.py

```diff
@@ -1,51 +1,339 @@
+
+    def _handle_message(self, topic_id, data) -> bool:
+        """ Handle text messages sent received on a topic  """
+        if (
+            self._stop_event.ready()
+        ):
+            # Ignore when stopped
+            return False
+
+        # TODO validate signature of the message
+
+        messages = list()  # TODO validate_and_parse_message(event["content"]["body"], peer_address)
+
+        if not messages:
+            return False
+
+        self.log.info(
+            "Incoming messages",
+            messages=messages,
+            topic_id=pex(topic_id),
+        )
+
+        for message in messages:
+            if not isinstance(message, (SignedRetrieableMessage, SignedMessage)):
+                self.log.warning("Received invalid message", message=message)
+            if isinstance(message, Delivered):
+                self._receive_delivered(message)
+            elif isinstance(message, Processed):
+                self._receive_message(message)
+            else:
+                assert isinstance(message, SignedRetrieableMessage)
+                self._receive_message(message)
+
+        return True
+
+    def _receive_delivered(self, delivered: Delivered):
+        self.log.info(
+            "Delivered message received", sender=pex(delivered.sender), message=delivered
+        )
+
+        assert self._raiden_service is not None
+        self._raiden_service.on_message(delivered)
+
+    def _receive_message(self, message: Union[SignedRetrieableMessage, Processed]):
+        assert self._raiden_service is not None
+        self.log.info(
+            "RIF Comms Message received",
+            node=pex(self._raiden_service.address),
+            message=message,
+            sender=pex(message.sender),
+        )
+
+        try:
+            delivered_message = Delivered(delivered_message_identifier=message.message_identifier)
+            self._raiden_service.sign(delivered_message)
+
+            queue_identifier = QueueIdentifier(
+                recipient=message.sender, channel_identifier=CHANNEL_IDENTIFIER_GLOBAL_QUEUE
+            )
+            self.send_message(*TransportMessage.wrap(queue_identifier, delivered_message))
+            self._raiden_service.on_message(message)
+
+        except (InvalidAddress, UnknownAddress, UnknownTokenAddress):
+            self.log.warning("Exception while processing message", exc_info=True)
+            return
 
     def start_health_check(self, address: Address):
-        raise NotImplementedError
+        self.log.debug("Healthcheck", peer_address=pex(address))
 
     def whitelist(self, address: Address):
-        raise NotImplementedError
+        self.log.debug("Whitelist", peer_address=pex(address))
 
     def link_exception(self, callback: Any):
-        raise NotImplementedError
+        self.greenlet.link_exception(callback)
 
     def join(self, timeout=None):
-        raise NotImplementedError
+        self.greenlet.join(timeout)
+
+    def listen_messages(
```

Comment:
for grammatical reasons i suggest we refactor → rename this to `listen_for_messages`

Reply:
done